samtranslator/translator/translator.py

```diff
@@ -268,6 +276,101 @@ def _get_resources_to_iterate(
 
         return functions + statemachines + apis + others + connectors
 
+    @staticmethod
+    def _update_resources(resources: Dict[str, Any], connectors_list: List[SamConnector]) -> None:
```

Comment:
Can make this pure (hence easier to reason and less error-prone) by returning the connectors as dict, and mutating at the caller level.